### ISSUE TITLE
Issue with JRuby

### ISSUE THREAD

**Issue by bryanvpham** [protocolbuffers/protobuf#3650]
Used `protoc` to generate `at.rb`.  My `main.rb` literally just includes `at.rb` and outputs "hello world". 

I built the gem from scratch following the instructions in the readme. Tried running my simple ruby program that 

```
RuntimeError: headers.key: "key" is already defined in "headers".
    build at com/google/protobuf/jruby/RubyDescriptorPool.java:84
   <main> at /root/work/at/at.rb:6
  require at org/jruby/RubyKernel.java:955
   (root) at uri:classloader:/jruby/kernel/kernel.rb:1
   <main> at main.rb:1
```

**Comment by pherl** [comment]
My impression is Jruby is currently broken and not maintained by anyone.. Adding @haberman for more info.

**Comment by felixcuello** [comment]
The workaround I found (for now) is to rename map structures and have unique names for all of them. Because as soon as I repeat the map name we get the exact same error.

**Comment by mjmateuci** [comment]
Do you guys found another way to solve this issue? @bryanvpham @pherl

**Comment by headius** [comment]
JRuby has been continuously maintained for over ten years and is still in very active use at many organizations. Can we get this fixed?

**Comment by acozzette** [comment]
At the moment I don't think we have anyone working on protobuf who is knowledgeable about JRuby unfortunately, but we would be glad to accept pull requests with fixes.

**Comment by jthack** [comment]
Pretty crazy this issue is still around.

**Comment by haberman** [comment]
I believe this is fixed in #7923.
